### PR TITLE
Issue 2534

### LINKED ISSUE #2534: Add Kotlinx Immutable collections
### Please confirm the following

- [X] I checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.

### What would you like to improve?

- Add https://github.com/Kotlin/kotlinx.collections.immutable
- Replace `List` usages with `ImmutableList` whenever you feel confident (optional)

### Because...

- Will greatly optimize the Compose code
- The CI will be happy
- Ivy Wallet transactions screens will run smoothly

### Description

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by mohammednawas8**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the **[Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md)**.
- [x] The code builds and is tested on a real Android device.
- [x] I confirm that I've run the code locally and everything works as expected.

## What's changed?
Describe with a few bullets **what's new:**
- a Refactored the viewModel List to ImmutableList

_💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!_

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b
- c

**In what cases your code won't work?**

- a
- b
- c

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #2534 

_Replace `{ISSUE_NUMBER}` with the id/number of the issue that you've fixed._

### CODE REVIEW

**Review by ILIYANGERMANOV — APPROVED** [review]
Excellent work 👏 Thank you! I'm very curious to see how app's performance has improved 💯
Before merging can you please make a screen recording with the APK from the "APK" GitHub Action to verify that the core functionality works. Try do add a few transactions and then scroll them.

File: gradle/libs.versions.toml
Comment (mohammednawas8) [reply gradle/libs.versions.toml:30]: okay

File: app/build.gradle.kts
Comment (mohammednawas8) [reply app/build.gradle.kts:171]: i will remove it

**Comment by mohammednawas8** [comment]
Uploading Untitled.mp4…

**Comment by mohammednawas8** [comment]
https://github.com/Ivy-Apps/ivy-wallet/assets/78867217/ad54d332-3013-4ddc-8a49-073119e55cc5

**Comment by ILIYANGERMANOV** [comment]
Great job! Seems like we're ready to merge after the required CI checks pass 🚀

**Pull request merged**
